pmlc/ast/builder.cc

```diff
@@ -738,6 +739,38 @@ struct ProgramBuilder {
         .result();
   }
 
+  Value makeGatherOp(ExprNodeIntrinsic *node, ArrayRef<Value> operands) {
+    TensorShape shape = evaluator.getShape(node);
+    RankedTensorType resultType = builder.getRankedTensorType(shape);
+    auto op = builder.create<tile::GatherOp>(
+        loc, resultType, ValueRange{operands[0], operands[1]},
+        ArrayRef<NamedAttribute>{});
+
+    FloatAttr cubicCoeff;
+    if (!m_Constant(&cubicCoeff).match(operands[4].getDefiningOp())) {
```

Comment:
use `matchPattern()` instead of `m_Constant().match()`. The `matchPattern()` takes an `Operation*` or `Value`. So it'd be like:
```
if (!matchPattern(operands[4], m_Constant(&cubicCoeff)) {
...
}
```

Reply:
Fixed